packages/app-desktop/gui/Sidebar/Sidebar.tsx

```diff
@@ -493,6 +493,20 @@ const SidebarComponent = (props: Props) => {
 	};
 
 	const renderAllNotesItem = (theme: Theme, selected: boolean) => {
+		const toggleAllNotesContextMenu = () => {
+			const menu = new Menu();
+
+			if (Setting.value('notes.perFolderSortOrderEnabled')) {
+				menu.append(new MenuItem({
+					...menuUtils.commandToStatefulMenuItem('togglePerFolderSortOrder', ALL_NOTES_FILTER_ID),
+					type: 'checkbox',
+					checked: PerFolderSortOrderService.isSet(ALL_NOTES_FILTER_ID),
+				}));
+			}
+
+			menu.popup({ window: bridge().window() });
+		};
```

Comment:
To avoid unnecessary rendering, callbacks should generally be created using `useCallback` so could you move the code to this?

Reply:
Right, now I understand the `useCallback` better, and makes sense in this case. I will do that.